### DISCUSSION TITLE
Unclear documentation for cassandra backend

### DISCUSSION THREAD

**Discussion by adrwong-lora** [celery/celery#8063] [General]
I was to use cassandra as the backend for my celery app.

I followed the documentation [here](https://docs.celeryq.dev/en/stable/userguide/configuration.html#std-setting-cassandra_servers) to update `app.conf` and provided an actual `cassandra.auth.PlainTextAuthProvider` class with _username_ and _password_ for the config field `cassandra_auth_provider`.

It failed with the `Authentication Failed` response from my cassandra client.

Then I walked through the source code, I saw the below lines
https://github.com/celery/celery/blob/6d13ec8f6033bc95bb9dbfeb1e3460394c59944d/celery/backends/cassandra.py#L127-L134

It turned out that `cassandra_auth_provider` both ` cassandra_auth_kwargs` have to be provided while `cassandra_auth_provider` actually accepts a string of either `PlainTextAuthProvider` or `SaslAuthProvider` instead of a class instance.

I successfully configured the backend afterwards.

Suggestion: Make the documentation more clear with expected class/type of the field or make the `cassandra_auth_provider` more flexible by allowing input as an actual class.

**Comment by auvipy** [comment]
can you contribute to improve this part?
